robyn/reloader.py

```diff
@@ -0,0 +1,84 @@
+import os
+import signal
+import subprocess
+import sys
+import time
+import psutil
+
+from watchdog.events import FileSystemEventHandler
+from watchdog.observers import Observer
+
+from robyn.logger import Colors, logger
+
+
+def setup_reloader(directory_path: str, file_path: str):
+    event_handler = EventHandler(file_path)
+
+    event_handler.reload()
+
+    logger.info(
+        f"Dev server initialized with the directory_path : {directory_path}",
+        Colors.BLUE,
+    )
+
+    def terminating_signal_handler(_sig, _frame):
+        event_handler.stop_server()
+        logger.info("Terminating reloader", bold=True)
+        observer.stop()
+        observer.join()
+
+    signal.signal(signal.SIGINT, terminating_signal_handler)
+    signal.signal(signal.SIGTERM, terminating_signal_handler)
+
+    observer = Observer()
+    observer.schedule(event_handler, path=directory_path, recursive=True)
+    observer.start()
+
+    try:
+        while observer.is_alive():
+            observer.join(1)
+    finally:
+        observer.stop()
+        observer.join()
+
+
+class EventHandler(FileSystemEventHandler):
+    def __init__(self, file_path: str) -> None:
+        self.file_path = file_path
+
+        self.last_reload = time.time()
+
+    def stop_server(self):
+        for process in psutil.Process().children(recursive=True):
+            process.kill()
+            process.wait()  # Required to avoid zombies
+
+    def reload(self):
+        self.stop_server()
+
+        new_env = os.environ.copy()
```

Comment:
and then check here.

Lmk, if this makes sense. I will be happy to push a commit to this PR otherwise 😅

Reply:
Also, if `multiprocessing.Event` doesn't work , we can have a look at this (https://docs.python.org/3/library/multiprocessing.html?highlight=event#multiprocessing.BoundedSemaphore) .

I haven't used this personally, but can give a deeper dive if need be.

Reply:
@sansyrox I'm really not convinced we can use those, with your method we would just create a new instance of `Event` each time we create a new process after reloading. I think we should merge the PR to have a first working solution, and maybe we could change things later? You could open a new PR if you manage to get something that works without env variables.